### ISSUE TITLE
[Bug]: Attributes do not save correctly

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#4016]
### Bug Description?

In the Item Form view when save is clicked, the save function first removes all attribute_links for that item_id, then saves the attributes as they have been entered from the form. This allows updates to the attributes without creating duplicates. The removal of the attribute_links is completing but the attributes are not being saved properly.

### Steps to Reproduce?

1. Edit an item or create a new item.
2. Add or edit an attribute.
3. Click submit.

### Expected Behavior?

old attribute_links to be removed and attributes as entered in the form to be saved with attribute_links

### OpensourcePOS Version

development (unreleased)

### Php version

Php 8.2

### What browsers are you seeing the problem on?

Firefox

### Server Operating System and version

Windows 11

### Database Management System and version

MySQL 8.2.0

### Web Server and version

Apache 2.4.58

### System Information Report (optional)

System Information Report

### Unmodified copy of OpensourcePOS

- [X] I agree this copy has not been modified

**Comment by jekkos** [comment]
Did a fix for this and tested both DATE and DRODPOWN attributes, they seem to save fine.

**Comment by objecttothis** [comment]
@jekkos I'm copying this from the Element conversation so that it's easy to find later.
There is definitely an issue with attributes that we will need to resolve. On a new database, I ran migrations, created 5 attributes (one of each type). Added TEXT, saved. Added DATE, saved. Added DROPDOWN, saved. The result is that it deleted the DATE and did not save the DROPDOWN value. When I've looked at the code in the past, the code first stores all the attributes (values) associated with the item in memory, then deletes the links. Finally it saves the new links. The problem is that somewhere after deleting the current links and saving the new ones, it fails, so data is lost. I was afraid this was an issue with existing databases, but this one is fresh. I think we need to re-open the issue for this.
